app/src/main/java/com/spendster/presentation/homeScreen/Dashboard/DashboardFragment.java

```diff
@@ -32,19 +35,27 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         return view;
     }
 
-    private void initRecyclerView(View view){
+    private void initRecyclerView(View view) {
         RecyclerView recyclerView = view.findViewById(R.id.expensesRecyclerView);
         recyclerView.setLayoutManager(new LinearLayoutManager(this.getContext()));
         recyclerView.setHasFixedSize(true);
         this.expenseRecyclerViewAdapter = new ExpenseRecyclerViewAdapter(new OnExpenseClickListener() {
             @Override
             public void onExpenseClick(Expense expense) {
-
+                showEditExpense(expense);
             }
         });
         recyclerView.setAdapter(this.expenseRecyclerViewAdapter);
     }
 
+    private void showEditExpense(Expense expense) {
+        Gson gson = new Gson();
+        String json = gson.toJson(expense);
+        Intent intent = new Intent(getContext(), EditExpensesActivity.class);
```

Comment:
@SimonMaha It's better to create a static function in EditExpensesActivity and pass parameters  there